### ISSUE TITLE
linter: `typescript/switch-exhaustiveness-check` silently ignores `defaultCaseCommentPattern`

### ISSUE THREAD

**Issue by zwang180** [oxc-project/oxc#24338]
### What version of Oxlint are you using?

1.70.0

### What command did you run?

`oxlint --type-aware --type-check test.ts`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["typescript"],
  "options": {
    "typeAware": true,
    "typeCheck": true
  },
  "rules": {
    "typescript/switch-exhaustiveness-check": [
      "error",
      {
        "allowDefaultCaseForExhaustiveSwitch": true,
        "considerDefaultExhaustiveForUnions": true,
        "defaultCaseCommentPattern": "^skip default$",
        "requireDefaultForNonUnion": false
      }
    ]
  }
}
```

### What happened?

**Disclosure: this issue was drafted with AI assistance and reviewed by me before submission, per the [AI Usage Policy](https://github.com/oxc-project/oxc/blob/main/CONTRIBUTING.md#ai-usage-policy)**

### Issue

The rule `typescript/switch-exhaustiveness-check` accepts `defaultCaseCommentPattern` in its configuration, but the option does NOT seem to be implemented by the `tsgolint` backend (`oxlint-tsgolint` 0.22.1). Matching comments are ignored and the switch is still reported as non-exhaustive.

### Minimal reproduction

```ts
declare const value: 'a' | 'b';

switch (value) {
  case 'a':
    break;
  // skip default
}
```

### Actual behavior

Oxlint reports:

```text
typescript(switch-exhaustiveness-check): Switch is not exhaustive. Cases not matched: "b"
```

### Expected behavior

The matching `// skip default` comment should be treated as an intentionally omitted default case, matching the configured pattern and upstream `typescript-eslint` behavior.

### Implementation evidence

The generated options type contains `DefaultCaseCommentPattern`, but the rule implementation never reads it.

The ported tests explicitly skip this behavior:

```go
// TODO(port): add support for DefaultCaseCommentPattern
Skip: true,
```

See [`switch_exhaustiveness_check_test.go`](https://github.com/oxc-project/tsgolint/blob/v0.22.1/internal/rules/switch_exhaustiveness_check/switch_exhaustiveness_check_test.go).

The option is also documented by [Oxlint](https://oxc.rs/docs/guide/usage/linter/rules/typescript/switch-exhaustiveness-check), so silently accepting but ignoring it is unexpected.

### Related issue

[#13500](https://github.com/oxc-project/oxc/issues/13500) fixed general option forwarding for this rule, but `defaultCaseCommentPattern` remains unimplemented.
